### PR TITLE
New efi combo updater

### PR DESCRIPTION
**Pull request by igor-stoppa**
Move back to shell script.
Handle also the alternate efi combo application used
for internal storage (after installation).

Signed-off-by: Igor Stoppa igor.stoppa@intel.com

### CODE REVIEW

**Comment by igor-stoppa** [comment]
I have tested this manually with a crafted new kernel file, but I was unable to test thoroughly with images because of various unrelated problems that came up wrt creation of swupd images.
I'm creating the PR so that it can be used while I'm OoO

**Comment by pohly** [comment]
@igor-stoppa how important is it to merge this? You mentioned on IRC that it fixes a known bug, and that there is a JIRA entry open for it. What is that bug number?

**Comment by igor-stoppa** [comment]
@pohly: I'm mostly concerned to make this available while I'll be away. I'll find a caretaker for it, but at least it's visible to everyone.
The issue it solves is https://jira01.devtools.intel.com/browse/IOTOS-1358

But it also handles the fact that now we have 2 efi combo applications: one for internal and one for external storage. That is needed, even if we do not have explicit requirement for it.

**Comment by igor-stoppa** [comment]
test this please

**Comment by igor-stoppa** [comment]
@avalluri @rojkov  this is the new updater

File: meta-ostro/recipes-swupd/swupd-client/swupd-client_%.bbappend [hunk 1]
```diff
@@ -23,16 +23,9 @@ SYSTEMD_SERVICE_${PN} += "${@ 'efi-combo-trigger.service' if ${OSTRO_USE_DSK_IMA
...
-    if [ "${OSTRO_USE_DSK_IMAGES}" = "True" ]; then
-        ${CC} ${LDFLAGS} ${WORKDIR}/efi_combo_updater.c  -Os -o ${B}/efi_combo_updater `pkg-config --cflags --libs glib-2.0`
-    fi
-}
-
 do_install_append () {
     if [ "${OSTRO_USE_DSK_IMAGES}" = "True" ]; then
-        install -d ${D}/usr/bin
-        install ${B}/efi_combo_updater ${D}/usr/bin/
+        install -m 0744 ${WORKDIR}/efi_combo_updater.sh ${D}/usr/bin/
```
Comment (avalluri) [inline meta-ostro/recipes-swupd/swupd-client/swupd-client_%.bbappend]: @igor-stoppa,  I think file name mismatch. so either 
update 'ExecStart=' in efi-combo-trigger.service to point to script.
or change the script name
install -m 0744 ${WORKDIR}/efi_combo_updater.sh ${D}/usr/bin/efi_combo_updater

Comment (igor-stoppa) [inline meta-ostro/recipes-swupd/swupd-client/swupd-client_%.bbappend]: Yes, htat's right. Thanks. I had fixed it locally, but forgot to refresh the patch.

**Comment by igor-stoppa** [comment]
test this please

**Comment by okartau** [comment]
test this please

**Pull request closed without merging**